### DISCUSSION TITLE
unknown service: esphome.nspanel_send_command_text_printf

### DISCUSSION THREAD

**Discussion by pkacp1** [Blackymas/NSPanel_HA_Blueprint#434] [General]
Im still fighting to make my NSPanel work with your project, but I can't succeed. I finally manages to upload TFT (needed to upload blank screen file first and then the correct one) and not I'm stuck with "Initializing.." screen. I double check all points mentioned here:

https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-Stuck-on-Initializing

but still won't work. My HA Services sends error message:

```
> NSPanel uses an unknown service
> The automation "NSPanel " (automation.nspanel) has an action that calls an unknown service: esphome.nspanel_send_command_text_printf.
> 
> This error prevents the automation from running correctly. Maybe this service is no longer available, or perhaps a typo caused it.
> 
> To fix this error, [edit the automation](http://homeassistant.local:8123/config/automation/edit/1678735592931) and remove the action that calls this service.
> 
> Click on SUBMIT below to confirm you have fixed this automation.
```

Anybody have any idea what's wrong?

**Comment by EBruse** [comment]
> Im still fighting to make my NSPanel work with your project, but I can't succeed. I finally manages to upload TFT (needed to upload blank screen file first and then the correct one) and not I'm stuck with "Initializing.." screen. I double check all points mentioned here:
> 
> https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-Stuck-on-Initializing
> 
> but still won't work. My HA Services sends error message:
> 
> ```
> > NSPanel uses an unknown service
> > The automation "NSPanel " (automation.nspanel) has an action that calls an unknown service: esphome.nspanel_send_command_text_printf.
> > 
> > This error prevents the automation from running correctly. Maybe this service is no longer available, or perhaps a typo caused it.
> > 
> > To fix this error, [edit the automation](http://homeassistant.local:8123/config/automation/edit/1678735592931) and remove the action that calls this service.
> > 
> > Click on SUBMIT below to confirm you have fixed this automation.
> ```
> 
> Anybody have any idea what's wrong?

I had the same error message with the automation configured for national language (Swedish). I changed to english and the error disappeared. Then i changed back to Swedish and now it works without the error.

**Comment by MichalAug** [comment]
I have the same with Polish. If I set English, the texts appear, but after changing to Polish they disappear.

**Comment by hollabee** [comment]
Did you get it solved?

> Im still fighting to make my NSPanel work with your project, but I can't succeed. I finally manages to upload TFT (needed to upload blank screen file first and then the correct one) and not I'm stuck with "Initializing.." screen. I double check all points mentioned here:
> 
> https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-Stuck-on-Initializing
> 
> but still won't work. My HA Services sends error message:
> 
> ```
> > NSPanel uses an unknown service
> > The automation "NSPanel " (automation.nspanel) has an action that calls an unknown service: esphome.nspanel_send_command_text_printf.
> > 
> > This error prevents the automation from running correctly. Maybe this service is no longer available, or perhaps a typo caused it.
> > 
> > To fix this error, [edit the automation](http://homeassistant.local:8123/config/automation/edit/1678735592931) and remove the action that calls this service.
> > 
> > Click on SUBMIT below to confirm you have fixed this automation.
> ```
> 
> Anybody have any idea what's wrong?

Did you get it solved?

**Comment by edwardtfn** [comment]
> Im still fighting to make my NSPanel work with your project, but I can't succeed. I finally manages to upload TFT (needed to upload blank screen file first and then the correct one) and not I'm stuck with "Initializing.." screen. I double check all points mentioned here:
> 
> https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-Stuck-on-Initializing
> 
> but still won't work. My HA Services sends error message:
> 
> ```
> > NSPanel uses an unknown service
> > The automation "NSPanel " (automation.nspanel) has an action that calls an unknown service: esphome.nspanel_send_command_text_printf.
> > 
> > This error prevents the automation from running correctly. Maybe this service is no longer available, or perhaps a typo caused it.
> > 
> > To fix this error, [edit the automation](http://homeassistant.local:8123/config/automation/edit/1678735592931) and remove the action that calls this service.
> > 
> > Click on SUBMIT below to confirm you have fixed this automation.
> ```
> 
> Anybody have any idea what's wrong?

Go to Developer Tools, select the tab "Services" and then type `send_command_text_printf`. It will show the service available containing this on its names:
![IMG_6117](https://user-images.githubusercontent.com/94725493/227731585-061d65f7-6d9a-42ea-9258-251e0f63c426.jpeg)

Could you please share the names you find?

**Comment by edwardtfn** [comment]
Anyone still having this issue or can I close this discussion?

**Comment by costakis79** [comment]
Yes I have still the issue

**Reply by edwardtfn** [reply]
Could you please [create an issue](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues) and provide more details there?
1. What is the error message? (A screenshot may help)
2. Which version of the TFT, ESPHome and Blueprint are shown in the boot screen in your panel?
3. Have you tried restarting Home Assistant?
4. Share your ESPHome settings (please hide your private info, like wifi credentials)
5. Etc...

**Reply by costakis79** [reply]
@edwardtfn it was a problem of my ESPHOME installation, it works fine

**Reply by SchreinerSe** [reply]
Hey,

I still have this problem.

![Screenshot 2023-09-21 220706](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/25008138/ccece609-bf9c-4545-9357-3bd9bcf66ad3)

TFT: 3.4
ESPHome: 4.0
Blueprint: 3.4

Home Assistant 2023.9.2

I am using HA in Docker on Synology.

Thanks 

Sebastian

**Reply by edwardtfn** [reply]
You have to update your TFT and Blueprint...

https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know#2-update-blueprint

**Reply by MrAshen** [reply]
@edwardtfn 

I am having the same issue. Freshly flashed the nspanel EU version
[logs_nspanelog_logs.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/14470117/logs_nspanelog_logs.txt)
, TFT and blueprint all are on 4.2.6. Two services are shown as unknown:

send_command_printf
page_home

Reimporting the blueprint and revalidating did not help.
![Screenshot_20221124_213318](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/104657242/f87aa094-a129-4179-9d97-f05de7ef7872)

**Reply by edwardtfn** [reply]
I believe this is due to a know issue we have with `friendly_name`.

- #1787 

If you have that set, could you please comment out for now?
You may have to reconnect your panel as described here: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1829#issuecomment-1959909591

If the error persists, please report as a new bug in the issues area, so we can investigate it deeper.

**Reply by MrAshen** [reply]
Yes, that seems to fix my error, I have commented it out and it works better now. By the way, where can I find the reboot option? there should be a config item in the lower left corner but it does not appear for me

**Reply by edwardtfn** [reply]
If you are looking for a reboot in the panel, you will se either in the boot screen (only after ESPHome establishes communication with the display) or on the settings page, which can be accessed by touching the time area for more than 1s.

**Reply by MrAshen** [reply]
Thanks @edwardtfn looks like that behavior changed compared to an earlier video.

By the way, shouldn't the button pages have a headline? They are just displayed in the entities pages.

**Reply by edwardtfn** [reply]
Yeah. A fix for that is also on the way. Hopefully we can release it later this week.

- #1814

**Reply by LeendertvdHoek** [reply]
Changing the friendly name to match the device name fixed it for me thkx!
